metadata/metadata.go

```diff
@@ -90,6 +90,20 @@ func Pairs(kv ...string) MD {
 	return md
 }
 
+// String implements the Stringer interface for pretty-printing a MD. Ordering of the values is non-deterministic as it ranges over a map.
+func (md MD) String() string {
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "MD{")
```

Comment:
Curious, Is there a reason for using `{` here? The go standard for printing maps seems to have `[` instead.

Reply:
The curly braces `{}` felt more natural for a map, and the regular braces `[]` felt more like an array to me. 
I guess because it feels more like json then, or more like structs?  But I used `=` instead of `:` too.

This is the current formatting in the PR:
```
"MD{k1=[v1, v2], k2=[], k3=[1, 2, 3]}"
``` 

It's true that when doing a `fmt.Println(m)` for `m` a map, it will print as `"map[k:v k:v]"`, would you prefer that format? 
```
"MD[k1:[v1, v2] k2:[] k3:[1, 2, 3]]"
```

Reply:
On second thoughts, i think this format is just fine. unless other maintainers have a strong opinion here. Thanks for clarifying.